src/CoreWCF.NetTcp/src/CoreWCF/Channels/ServerFramingDuplexSessionChannel.cs

```diff
@@ -215,8 +215,8 @@ private async ValueTask<(Message, ReadOnlySequence<byte>)> DecodeMessageAsync(Re
                                 Message message;
                                 try
                                 {
-                                    message = _connection.MessageEncoder.ReadMessage(
-                                        new ArraySegment<byte>(_connection.EnvelopeBuffer.ToArray(), 0, _connection.EnvelopeSize),
+                                    message = await _connection.MessageEncoder.ReadMessageAsync(
+                                        new ReadOnlySequence<byte>(_connection.EnvelopeBuffer.ToArray(), 0, _connection.EnvelopeSize),
                                         _connection.BufferManager,
```

Comment:
Calling ToArray creates a new array and copies the contents of the EnvelopeBuffer to that array. EnvelopeBuffer is a Memory\<byte\> which can be passed to the constructor of ReadOnlySequence as-is